File: LeetCode/2574_Left_and_Right_Sum_Differences/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
    let count=0;
    let left=0;
    let right=0;
    let leftArr=[];
    let rightArr = new Array(nums.length)
    while(count<nums.length){
        leftArr.push(left)
        rightArr[nums.length-1-count] = right
        left+=nums[count];
        right+=nums[nums.length-1-count];
        count++;
    }
    for(let i=0;i<leftArr.length;i++){
        leftArr[i] = Math.abs(leftArr[i]-rightArr[i])
    }
    return leftArr;
};

File: LeetCode/2574_Left_and_Right_Sum_Differences/solution.cpp
class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        int sum = 0;
        vector<int> left;
        for(int x:nums){
            left.push_back(sum);
            sum+=x;
        }
        for(int i = 0;i<nums.size();i++){
            int a = abs(sum-nums[i]-left[i]-left[i]);
            left[i]=a;
        }
        return left;
    }
};